packages/vsx-registry/src/browser/vsx-extensions-model.ts

```diff
@@ -232,30 +232,52 @@ export class VSXExtensionsModel {
                 if (!allVersions) {
                     continue;
                 }
-                const res = await client.query({ extensionId: id, extensionVersion: allVersions.version, includeAllVersions: true });
-                let verified = res.extensions?.[0].verified;
-                if (!verified) {
-                    if (res.extensions?.[0].publishedBy.loginName === 'open-vsx') {
-                        verified = true;
-                    }
-                }
-                if (!this.preferences.get('extensions.onlyShowVerifiedExtensions') || verified) {
-                    this.setExtension(id).update(Object.assign(data, {
-                        publisher: data.namespace,
-                        downloadUrl: data.files.download,
-                        iconUrl: data.files.icon,
-                        readmeUrl: data.files.readme,
-                        licenseUrl: data.files.license,
-                        version: allVersions.version,
-                        verified: verified
-                    }));
-                    searchResult.add(id);
+                if (this.preferences.get('extensions.onlyShowVerifiedExtensions')) {
+                    const verified = await this.fetchVerifiedStatus(id, client, allVersions);
```

Comment:
I don't think we have to `await` the `fetchVerifiedStatus` here: it will serialize the requests to openVSX. Why not 
```
this.fetchVerifiedStatus(id, client, allVersions).then(verified => {
                        this.doChange(() => {
                            this.updateExtensions(data, id, allVersions, !!verified);
                            return Promise.resolve();
                        });
                    })
```
We will fire lots of parallel requests to OpenVSX, but we're also doing that in the other branch of the `if` anyway.

Reply:
Good point, that works better !